File: num124.java
import java.util.Stack;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

class num124 {
    public int maxPathSum(TreeNode root) {
        int max = Integer.MIN_VALUE;
        Stack<TreeNode> stack = new Stack<>();
        if(root != null){
            stack.push(root);
            TreeNode c = null;
            while(!stack.isEmpty()){
            c = stack.peek();
            if(c.left != null && root != c.left && root != c.right){
                stack.push(c.left);
            }else if(c.right != null && root != c.right){
                stack.push(c.right);
            }else{
                TreeNode peek = stack.pop();
                int result = peek.val;
                if(peek)
                root = c;
            }
        }
    }

File: num124.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def maxPathSum(self, root: TreeNode) -> int:
        self.res = -float('inf')
        def rootSum(node):
            if not node:
                return 0
            ls = rootSum(node.left)
            rs = rootSum(node.right)
            self.res = max(self.res, node.val+max(ls+rs, ls, rs, 0))
            if ls >= rs and ls > 0:
                return node.val + ls
            if rs >= ls and rs > 0:
                return node.val + rs
        
        rootSum(root)
        return self.res

if __name__ == "__main__":
    pass